### PR TITLE
fixes #5811 - Validates emails to RFC 5322 specification

### PR DESCRIPTION
**Pull request by cfouant**

### CODE REVIEW

**Comment by ohadlevy** [comment]
@cfouant thanks, please see test failures.

**Comment by cfouant** [comment]
@ohadlevy - all fixed, thanks!

File: test/unit/user_test.rb [hunk 1]
```diff
@@ -299,10 +299,24 @@ def setup_user operation
   end
 
   test "email with whitespaces should be stripped" do
-    u = User.create! :auth_source => auth_sources(:one), :login => "boo", :mail => "b oo@localhost "
+    u = User.create! :auth_source => auth_sources(:one), :login => "boo", :mail => " boo@localhost "
```
Comment (dLobatog) [inline test/unit/user_test.rb]: Why the change on this test?

File: test/unit/user_test.rb [hunk 2]
```diff
@@ -299,10 +299,24 @@ def setup_user operation
...
     assert_equal u.mail, "boo@localhost"
   end
 
+  test "email should not have special characters outside of quoted string format" do
+    u = User.new :auth_source => auth_sources(:one), :login => "boo", :mail => "specialchars():;@example.com"
```
Comment (dLobatog) [inline test/unit/user_test.rb]: Please use meaningful variable names, in this case something like "user" would be :ok_hand: 
Also this would be better as two separate tests, it'll help us figure out faster which one fails.

**Comment by dLobatog** [comment]
This looks ready to merge after the small fixes commented in the code.

File: test/unit/user_test.rb (hunk 1 above)
Comment (cfouant) [inline test/unit/user_test.rb]: Since spaces are allowed within an email using the quoted-string form, it doesn't make sense to strip internal whitespaces anymore.

**Comment by cfouant** [comment]
@eLobato - comments and changes made.  Thanks!

**Comment by dLobatog** [comment]
Awesome, will merge if tests are :green_book:

**Comment by dLobatog** [comment]
Merged as https://github.com/theforeman/foreman/commit/bd6b42715d2052c99f285dac9b919c27b36453a4 for Foreman 1.7 , thanks @cfouant !
